<launch>
  <arg name="chip" default="gpiochip0"/>
  <arg name="offset_pushbutton" default="0"/>
  <arg name="display_survey_status" default="false"/>
  <arg name="offset_survey_status" default="0"/>
  <arg name="distance_threshold" default="1.0"/>
  <arg name="active_low" default="false"/>
  <arg name="port" default="/dev/ttyACM0"/>
  <arg name="baud" default="57600"/>
  <arg name="neopixel_rosserial" default="true" />
  <arg name="neopixel_python" default="false" />
  <arg name="output" default="log" />
  <arg name="ns" default="/piksi_multi_cpp_base/base_station_receiver_0"/>

  <!-- Neopixel settings -->
  <arg name="brightness" default="1.0"/>

  <group ns="$(arg ns)">
    <node name="pushbutton" pkg="piksi_multi_interface" type="pushbutton" output="$(arg output)" required="true">
      <param name="rate" value="20"/>
      <!-- Find chip and line with gpiod cli gpiodetect -->
      <param name="chip" value="$(arg chip)"/>
      <param name="offset" value="$(arg offset_pushbutton)"/>
      <param name="active_low" value="$(arg active_low)"/>
      <!--param name="num_desired_fixes" value="100"/-->
      <!--param name="offset_z" value="2.0"/-->
    </node>

    <group if="$(eval arg('display_survey_status'))">
      <node name="survey_status" pkg="piksi_multi_interface" type="survey_status" output="$(arg output)" required="true">
        <param name="chip" value="$(arg chip)"/>
        <param name="offset" value="$(arg offset_survey_status)"/>
        <param name="distance_threshold" value="$(arg distance_threshold)"/>
      </node>
    </group>

    <group if="$(eval arg('neopixel_rosserial'))">
      <node name="piksi_status_led" pkg="rosserial_server" type="serial_node" output="$(arg output)" required="true">
        <param name="port" value="$(arg port)"/>
        <param name="baud" value="$(arg baud)"/>
      </node>
    </group>

    <group if="$(eval arg('neopixel_python'))">
      <node name="neopixel_node" pkg="piksi_multi_interface" type="neopixel_node.py" output="$(arg output)" required="true">
        <param name="brightness" value="$(arg brightness)"/>
      </node>
    </group>
  </group>
</launch>
